util/configurations.py

```diff
@@ -144,6 +145,33 @@ def _get_pv_name_without_field(pv_name):
         else:
             return pv_name
 
+    def get_devices_directory(self):
+        raise NotImplementedError("This is an abstract class, use a concrete class instead")
+
+    def get_device_screens_from_xml(self):
+        """
+        Gets the XML corresponding to the Device Screens for the instrument
+        Returns: the XML as a string
+        """
+        path = os.path.join(self.get_devices_directory(), "screens.xml")
+        with open(path) as xml_file:
+            return xml_file.read()
+
+    def get_device_screens(self, xml):
+        """
+        Args:
+            xml: XML inout to parse
+        Returns:
+            device_screens (list): List of device screens
+        """
+        root = ET.fromstring(xml)
+
+        device_screens = []
+        for device in root:
+            device_screens.append(device.find(f"{self.DEVICES_XML_SCHEMA}key").text)
+
+        return set(device_screens)
```

Comment:
up to you whether you think this is nicer, but you can do this with set comprehension

```suggestion
        return {device.find(f"{self.DEVICES_XML_SCHEMA}key").text for device in root}
```